### PR TITLE
gixy: use toPythonApplication

### PR DESCRIPTION
**Pull request by misuzu**
###### Motivation for this change
This fixes python issues in nix-shell when using non-default python version.

###### Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [X] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [X] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - [X] `nixosTests.nginx`
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/gixy/default.nix [hunk 1]
```diff
@@ -1,11 +1,19 @@
...
+, isPy27
+, nose
+, cached-property
+, configargparse
+, jinja2
+, pyparsing
+, six }:
 
-python3.pkgs.buildPythonApplication rec {
+buildPythonApplication rec {
```
Comment (FRidh) [inline pkgs/development/python-modules/gixy/default.nix:11]: buildPythonPackage

**Review by FRidh — CHANGES_REQUESTED** [review]
> This fixes python issues in nix-shell when using non-default python version.

Could you further explain what you mean here?

**Comment by misuzu** [comment]
> > This fixes python issues in nix-shell when using non-default python version.
> 
> Could you further explain what you mean here?

For example, if I want to use `python36` in my `shell.nix` and then I also add `gixy` I'll have two interpreters in my `PATH`. This leads to really hard to debug issues, especially with C-extensions.

**Comment by FRidh** [comment]
Yes, that's definitely an annoying issue. But this is not the way to fix it. The core issue you see is that the interpreter is propagated, but that's done for convenience when building a package.

I don't know your way of working. I take it you do something like
```nix
mkShell {
  nativeBuildInputs = [
    gixy 
    python36
  ];
}
```

What happens is that you essentially get the build environment for the shell, and that includes the propagated Python because `nix-shell` runs build hooks. What you actually want, is just the packages themselves. If you use the new `nix` command you get that with `nix shell`. Using the old `nix-shell`, you need to set `PATH` somehow, e.g. using

```nix
with import (fetchTarball "channel:nixos-unstable") {};

let
  packages = [
    gixy
    python36
  ];

in mkShell {
  shellHook = ''
    export PATH=${lib.makeBinPath packages}:$PATH
  '';
}
```
Note I am not using here the `packages` parameter to `mkShell` because that's just an alias for `*Inputs` and doesn't solve the issue.

**Pull request closed without merging**